### PR TITLE
Enhance model output type

### LINKED ISSUE #175: Expand Type for Results in Predictions
Right now the type of `results` in `predictions` is a `string` -- this has served us well for now, but will need to be expanded because we expect that models may have more than result that end users care about (for instance a COG, geojson, map layer).

### Acceptance Criteria

This issue is to change `results` to an array of `StacAsset` objects. Signing of S3 URLs should behave the same.

### PR DESCRIPTION
**Pull request by jisantuc**
## Overview

This PR makes models accept a list of STAC assets instead of a single string for output. It migrates the db column to a non-nullable `jsonb` field instead of a string and updates the types in the api.

### Checklist

- [x] Any new migrations have been audited to make sure they won't kill the application while being applied

## Testing Instructions

- assemble API
- start up server
- post the mock ml training model to `:8080/api/models` because its json schema is really easy to satisfy:

```json
{
  "name": "Mock ML Training",
  "validator": {
    "schema": {
      "$id": "http://example.com/example.json",
      "type": "object",
      "title": "The Root Schema",
      "$schema": "http://json-schema.org/draft-07/schema",
      "default": {},
      "required": [
        "NUM_EPOCHS",
        "TRAIN_SIZE",
        "ROOT_URI"
      ],
      "properties": {
        "ROOT_URI": {
          "$id": "#/properties/ROOT_URI",
          "type": "string",
          "title": "The Root_uri Schema",
          "default": "",
          "examples": [
            "s3://foo/bar/"
          ],
          "description": "An explanation about the purpose of this instance."
        },
        "NUM_EPOCHS": {
          "$id": "#/properties/NUM_EPOCHS",
          "type": "integer",
          "title": "The Num_epochs Schema",
          "default": 0,
          "examples": [
            3
          ],
          "description": "An explanation about the purpose of this instance."
        },
        "TRAIN_SIZE": {
          "$id": "#/properties/TRAIN_SIZE",
          "type": "integer",
          "title": "The Train_size Schema",
          "default": 0,
          "examples": [
            100
          ],
          "description": "An explanation about the purpose of this instance."
        }
      },
      "description": "The root schema comprises the entire JSON document.",
      "additionalProperties": true
    }
  },
  "jobDefinition": "mock-ml-train:2",
  "jobQueue": "granaryJobQueueManual"
}
```

- create a prediction for it on `:8080/api/predictions` with:

```json
{
  "modelId": "<your model id>",
  "arguments": {"ROOT_URI": "", "NUM_EPOCHS": 0, "TRAIN_SIZE": 0}
}
```

- get its webhook id from the response or the database
- update its results by posting to `:8080/api/predictions/<prediction id>/results/<webhook id>`:

```
{
  "results": [
    {
      "href": "lol",
      "type": "vendor-media-type",
      "roles": [
        "data",
        "vendor role"
      ],
      "title": "lol",
      "description": "lol"
    }
  ]
}
```

- inspect the prediction you get back and note that stac4s didn't mangle the vendor fields as well -- not something that was accomplished here, but nice anyway

Closes #175

### CODE REVIEW

**Comment by jisantuc** [comment]
Woops yup https://raster-foundry.github.io/granary/docs/introduction#whats-a-prediction

**Comment by jisantuc** [comment]
I also included colors / illustrations in this PR, since I already had docs site changes going.

File: database/src/main/resources/migrations/V7__make_stac_links_of_existing_output.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
+ALTER TABLE predictions ADD COLUMN output_location_copy jsonb not null default '[]';
+
+UPDATE predictions SET output_location_copy = CASE
+    WHEN right(output_location, 7) = 'geojson'
```
Comment (notthatbreezy) [inline database/src/main/resources/migrations/V7__make_stac_links_of_existing_output.sql:4]: I didn't know about this function - nice

File: website/build/granary/index.html [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+</span></div></h2><div><span><p>Associating data with the model run that produced it can be hard. With Granary tracking the arguments, outputs, and infrastructure for your tasks, you won't have to guess which configuration produced which results.</p>
+</span></div></div></div></div></div></div><div class="container darkBackground paddingBottom paddingTop"><div class="wrapper"><div class="gridBlock"><div class="blockElement imageAlignSide imageAlignRight twoByGridBlock"><div class="blockContent"><h2><div><span><p>Stop kicking off doomed models</p>
 </span></div></h2><div><span><p>Granary puts some guardrails in front of your interaction with Batch so you don't fall in to common mistakes, like mis-formatted arguments,missing arguments, and bad string values for important AWS resources.</p>
```
Comment (notthatbreezy) [inline website/build/granary/index.html:10]:
```suggestion
</span></div></h2><div><span><p>Granary puts some guardrails in front of your interaction with Batch so you don't fall in to common mistakes, like mis-formatted arguments, missing arguments, and bad string values for important AWS resources.</p>
```

Just noticed this space is missing

File: database/src/main/resources/migrations/V7__make_stac_links_of_existing_output.sql (hunk 1 above)
Comment (jisantuc) [reply database/src/main/resources/migrations/V7__make_stac_links_of_existing_output.sql:4]: me neither -- i dug through the string functions page looking for a `last` or something and got lucky :game_die:

**Pull request merged**